schemas/user.schema.ts

```diff
@@ -0,0 +1,33 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document, Types } from 'mongoose';
+
+export type UserDocument = User & Document<Types.ObjectId>;
+
+export type UserDocumentWithoutPassword = Omit<User, 'password'> &
+  Document<Types.ObjectId>;
+
+@Schema()
+export class User {
+  @Prop({ required: true })
+  first_name: string;
```

Comment:
https://www.mongodb.com/community/forums/t/does-mongodb-have-a-naming-convention-for-collection-and-field-naming/11806/2 I would suggest using camelCase naming convention like in JS/TS

Reply:
Just to be clear, when using PostgreSQL the convention is to use snake case and with Mongo it is camel case, right?

Reply:
Actually, there is no strong convention for any of these dbs. With C#, for example, we are using PascalCase. I would suggest to use convenient case for your programming language